src/Propel/Runtime/Util/Profiler.php

```diff
@@ -213,7 +213,11 @@ public function isSlow()
      */
     public function getProfile()
     {
-        return $this->getProfileBetween($this->snapshot, self::getSnapshot());
+        $endSnapshot = self::getSnapshot();
+        $startSnapshot = (empty($this->snapshot)) ? $endSnapshot : $this->snapshot;
```

Comment:
```suggestion
        $startSnapshot = ($this->snapshot === null) ? $endSnapshot : $this->snapshot;
```
?
when you make the empty case the null one?